vision/src/autogluon/vision/image_prediction/image_prediction.py

```diff
@@ -0,0 +1,319 @@
+                train_data, val_data = train_data.split(1 - holdout_frac)
+        if isinstance(val_data, str):
+            from d8.image_classification import Dataset as D8D
+            names = D8D.list()
+            if val_data.lower() in names:
+                val_data = D8D.get(val_data)
+            else:
+                valid_names = '\n'.join(names)
+                raise ValueError(f'`val_data` {val_data} is not among valid list {valid_names}')
+        if self._classifier is not None:
+            logging.getLogger("ImageClassificationEstimator").propagate = True
+            self._classifier._logger.setLevel(log_level)
+            self._fit_summary = self._classifier.fit(train_data, val_data, 1 - holdout_frac, random_state, resume=False)
+            return
+
+        # new HPO task
+        config={'log_dir': self._log_dir,
+                'num_trials': 99999 if num_trials is None else max(1, num_trials),
+                'time_limits': time_limit,
+                'search_strategy': search_strategy,
+                }
+        if nthreads_per_trial is not None:
+            config.update({'nthreads_per_trial': nthreads_per_trial})
+        if ngpus_per_trial is not None:
+            config.update({'ngpus_per_trial': ngpus_per_trial})
+        if dist_ip_addrs is not None:
+            config.update({'dist_ip_addrs': dist_ip_addrs})
+        if epochs is not None:
+            config.update({'epochs': epochs})
+        if isinstance(hyperparameters, dict):
+            net = hyperparameters.pop('net', None)
+            if net is not None:
+                config.update({'custom_net': net})
+            optimizer = hyperparameters.pop('optimizer', None)
+            if optimizer is not None:
+                config.update({'custom_optimizer': optimizer})
+            # check if hyperparameters overwriting existing config
+            for k, v in hyperparameters.items():
+                if k in config:
+                    raise ValueError(f'Overwriting {k} = {config[k]} to {v} by hyperparameters is ambiguous.')
+            config.update(hyperparameters)
+        if scheduler_options is not None:
+            config.update(scheduler_options)
+        if use_rec == True:
+            config['use_rec'] = True
+        # verbosity
+        if log_level > logging.INFO:
+            logging.getLogger('gluoncv.auto.tasks.image_classification').propagate = False
+            logging.getLogger("ImageClassificationEstimator").propagate = False
+            logging.getLogger("ImageClassificationEstimator").setLevel(log_level)
+        task = _ImageClassification(config=config)
+        task._logger.setLevel(log_level)
+        task._logger.propagate = True
+        self._classifier = task.fit(train_data, val_data, 1 - holdout_frac, random_state)
+        self._classifier._logger.setLevel(log_level)
+        self._classifier._logger.propagate = True
+        self._fit_summary = task.fit_summary()
+
+    def predict_proba(self, x):
```

Comment:
There is still an output format difference here for both predict and predict_proba compared to Tabular. I'm ok with it for this PR but we probably want to align before v0.1 release.